### PR TITLE
Fix small typo in fungible token freeze docs

### PR DESCRIPTION
**Pull request by tsenovilla**
Hi!

In the course of a talk with @shawntabrizi in Singapore, we realized the documentation related to freeze balances' a little bit confusing. It stated that a frozen amount is released at some specified block number, which isn't true in general. 

This PR fixes that typo and further specifies that the frozen balance may exceed the available balance, according to what we learned at the PBA. This feature was not specified in the documentation AFAIK.

This is the first time I submit something to the polkadot SDK repo, so please feel free to rephrase the docs I added in case I messed up!

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]
lgtm. thanks!

File: substrate/frame/support/src/traits/tokens/fungible/mod.rs [hunk 1]
```diff
@@ -58,8 +58,9 @@
 //!   3 holds for 100 units, the account can spend its funds for any reason down to 300 units, at
 //!   which point the holds will start to come into play.
 //!
-//! - **Frozen Balance**: A freeze on a specified amount of an account's free balance until a
-//!   specified block number.
+//! - **Frozen Balance**: A freeze on a specified amount of an account's balance. There's no 
+//!   restriction preventing an account from freezing over its available balance. This amount still
+//!   belongs to the account holder, but cannot be spent.
```
Comment (ggwpez) [inline substrate/frame/support/src/traits/tokens/fungible/mod.rs:63]: >This amount still belongs to the account holder, but cannot be spent.

Can you be more specific please? Since the frozen balance can be over the total balance, it does not belong to the account. If an account has 10 total balance and 20 frozen, the 20 does not belong to the account, just the lock does.

Comment (shawntabrizi) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs:63]: fair point

Comment (tsenovilla) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs:63]: Definitely! Let me rephrase it then :) Should I close this PR and create a new one ?

Comment (shawntabrizi) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs:63]: > Should I close this PR and create a new one ?

def no, just make the fix and commit the changes

Comment (tsenovilla) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs]: @shawntabrizi  @ggwpez  done! I hope now it's better explained :)

File: substrate/frame/support/src/traits/tokens/fungible/mod.rs [hunk 2]
```diff
@@ -58,13 +58,17 @@
 //!   3 holds for 100 units, the account can spend its funds for any reason down to 300 units, at
 //!   which point the holds will start to come into play.
 //!
-//! - **Frozen Balance**: A freeze on a specified amount of an account's free balance until a
-//!   specified block number.
+//! - **Frozen Balance**: A freeze on a specified amount of an account's balance. This amount is 
+//!   set aside and cannot be spent by the account holder.
```
Comment (shawntabrizi) [inline substrate/frame/support/src/traits/tokens/fungible/mod.rs:62]:
```suggestion
//! - **Frozen Balance**: A freeze on a specified amount of an account's balance. Tokens which 
//!   are frozen cannot be transferred.
```

I think it is best to state things directly. This is I think the most obvious and accurate statement about frozen balances, and does not contradict other ways of manipulating balances like holds.

Comment (tsenovilla) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs:62]: > I think it is best to state things directly. This is I think the most obvious and accurate statement about frozen balances, and does not contradict other ways of manipulating balances like holds.

Yeah! It's clearer that way. I commit it

File: substrate/frame/support/src/traits/tokens/fungible/mod.rs [hunk 3]
```diff
@@ -58,13 +58,17 @@
...
+//!   set aside and cannot be spent by the account holder.  
 //!
 //!   Multiple freezes always operate over the same funds, so they "overlay" rather than
 //!   "stack". This means that if an account has 3 freezes for 100 units, the account can spend its
 //!   funds for any reason down to 100 units, at which point the freezes will start to come into
 //!   play.
+//!   
+//!   It's important to note that the frozen balance can exceed the total balance of the account.
+//!   In such cases, the account cannot spend any fund and the excess frozen amount doesn't 
+//!   actually belong to the account, only the lock does.
```
Comment (shawntabrizi) [inline substrate/frame/support/src/traits/tokens/fungible/mod.rs]:
```suggestion
//!   It's important to note that the frozen balance can exceed the total balance of the account.
//!   This can be used in scenarios where you would want to keep an account's balance frozen, independent of the specific amount of balance they have. For example, by setting the users frozen balance to `Balance::MAX`, the user will not be able to transfer any funds in the account. Developers should be conscious to use the `balance_freezable` function to determine the amount of available balance of the user which can become frozen.
```

File: substrate/frame/support/src/traits/tokens/fungible/mod.rs [hunk 4]
```diff
@@ -58,13 +58,17 @@
...
+//!   are frozen cannot be transferred.
 //!
 //!   Multiple freezes always operate over the same funds, so they "overlay" rather than
 //!   "stack". This means that if an account has 3 freezes for 100 units, the account can spend its
 //!   funds for any reason down to 100 units, at which point the freezes will start to come into
 //!   play.
+//!   
+//!   It's important to note that the frozen balance can exceed the total balance of the account.
+//!   In such cases the excess frozen amount doesn't actually belong to the account, only the lock
+//!   does.
```
Comment (shawntabrizi) [inline substrate/frame/support/src/traits/tokens/fungible/mod.rs:71]:
```suggestion
//!   It's important to note that the frozen balance can exceed the total balance of the account.
//!   This can be used in scenarios where you would want to keep an account's balance frozen, independent of the specific amount of balance they have. For example, by setting the users frozen balance to `Balance::MAX`, the user will not be able to transfer any funds in the account. Developers should be conscious to use the `balance_freezable` function to determine the amount of available balance of the user which can become frozen.
```

Here is some un-formatted word-vomit.

Perhaps give it a read yourself and write it in your own words which make sense to a new user.

Comment (tsenovilla) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs:71]: Awesome, I've to leave now but I'll rephrase it asap. Having an use example is helpful cause the idea isn't natural at all 👍

Comment (tsenovilla) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs]: > Here is some un-formatted word-vomit.
> 
> Perhaps give it a read yourself and write it in your own words which make sense to a new user.

I didn't include finally the mention to the `balance_freezable` function as I think it may lead to confusion anyway. The amount specified in that function may be bypassed using `Fortitude::Force`, so saying in the same sentence that there's no upper bound in the freezable amount and that developers may use this function to set such a bound feels weird IMO. Anyway, the purpose of that function's well explained in its own doc

Comment (shawntabrizi) [reply substrate/frame/support/src/traits/tokens/fungible/mod.rs]: all good to me

**Review by shawntabrizi — APPROVED** [review]

**Comment by ggwpez** [comment]
Can you still please put a minimal [prdoc](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/prdoc.md) and record a `patch` for `frame-support`? Otherwise it will possibly not be included in the next release.

**Comment by tsenovilla** [comment]
> Can you still please put a minimal [prdoc](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/prdoc.md) and record a `patch` for `frame-support`? Otherwise it will possibly not be included in the next release.

Sure! I wasn't aware of this procedure, let me do it later on

File: prdoc/pr_4943.prdoc
Comment (tsenovilla) [inline prdoc/pr_4943.prdoc:1]: @ggwpez would this be a correct prdoc?

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot fmt
